File: 007/Problem7.py
import time


def isPrime(num: int) -> bool:
    for i in range(2, num):
        if num % i == 0:
            return False
    return True


def problem():
    count = 2
    i = 4
    while True:
        i += 1
        if (isPrime(i)):
            count += 1
            if count == 10_001:
                break
    print(i)


start = time.time()
for i in range(10):
    problem()
end = time.time()
print("Function took " + str(end - start))

File: 007/Problem7.rs
fn main() {
    let start = std::time::SystemTime::now();
    for _ in 0..10 {
        problem();
    }
    match start.elapsed() {
        Ok(elapsed) => {
            println!("Function took: {}", elapsed.as_nanos())
        }
        Err(e) => {
            println!("Error: {:?}", e)
        }
    }
}

fn prime(num: i64) -> bool {
    for i in 2..num {
        if num % i == 0 {
            return false;
        }
    }
    return true;
}

fn problem() {
    let mut count = 2;
    let mut i = 4;
    loop {
        i += 1;
        if prime(i) {
            count += 1;
            if count == 10_001 {
                break;
            }
        }
    }
    println!("{}", i);
}
